ls1intum/Artemis
# Add programming assessment dashboard

### Checklist

### Motivation and Context


Unlike modeling and textexercises, there was no assessment-dashboard for programming exercises. In [#2745](https://github.com/ls1intum/Artemis/pull/2745) we adapted the dashboards for the second correction. The dashboard we create here should work in the same way as those.

### Description
You can access the dashboard from the exercise groups page, when you click on "Submission".
![image](https://user-images.githubusercontent.com/33342534/107053826-958eff80-67cf-11eb-9d63-1c92c4da7c1a.png)
There you will see a similar dashboard as with other exercise types.
![image](https://user-images.githubusercontent.com/33342534/107431935-7a9df180-6b27-11eb-9f62-9ef84c9e6bab.png)



### Steps for Testing


1. create an exam with 2 correction rounds, and with a programming exercise
2. participate with a few students (2-5)

Now, the easiest way to test, is with two users simultaneously in two different browser windows. With one tutor account go to the exercise-assessment dashboard. With the instructor account go to the new assessment dashboard. (exam -> exercise-groups -> programming-exercise -> submissions)

First the assessment dashboard should show submissions only with automatic results. Now start an assessment with the tutor from the exercise-assessment-dashboard.
 <img src="https://user-images.githubusercontent.com/33342534/107117648-11df1c80-687c-11eb-8973-48cfbc01b821.png" width="350">

Refresh the assessment-dashboard, now one locked submission should be visible. Submit with the tutor, now it should be visible with "open assessment" from the instructors perspective. When you open it with the instructor, make sure everything is loaded correctly.

With a different tutor do the second correction, and test the same things again. (You might need to enable the second correction with the instructor in the exercise-assessment-dashboard).

Now use the assessment dashboard with the instructor to assess different submissions in the first/second round.

When opening results with "open/continue assessment" be sure that you get the result that you would expect.

### Test Coverage



programming-assessment-dashboard.component.ts: 0% (Not existing yet, I will create it before the merge)
